### ISSUE TITLE
[bug] Gemma4 Dense model lacks final logit softcapping

### ISSUE THREAD

**Issue by pavelgein** [NVIDIA-NeMo/Megatron-Bridge#4610]
### Problem

In the [request](https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/4148) final logit softcapping was added only for MoE models

### Minimal repro

```shell
Run Gemma4
```

### Expected behavior

Logits are capped

### Affected area

area:model

### Regression?

Yes

### Logs

```shell

```
